lib/resolution/resolver.js

```diff
@@ -63,8 +63,7 @@ class Resolver {
      * @returns {boolean} whether the type is configured to be optional
      */
     isOptional(type) {
-        // TODO temporary solution to determine optional parameters. Align w/ compiler/importer.
-        return Object.keys(type).some(k => k.startsWith('@Core.OptionalParameter'))
+        return !type.notNull
```

Comment:
Hi @daogrady & @chgeo 👋 🙂  

I found out a small bug in current `cds-typer@0.27.0` version that CDS array-like params are always compiled into optional TS props.

_A quick example to reproduce_:
```cds
function check(
    @mandatory param_1: String not null,
    @mandatory param_2: Array of String not null
) returns ...
```

_which leads to wrong compiled TS props_:
```TypeScript
export declare const check: {
    ...
    __parameters: { param_1: string; param_2?: string[] };
    ...
}
```

The hotfix is bellow:
```suggestion
        return type.items ? !type.items.notNull : !type.notNull
```

Reply:
Thanks for catching this @siarhei-murkou.  Will fix it.